Client/Frontend/Browser/Playlist/Cache/PlaylistCacheLoader.swift

```diff
@@ -0,0 +1,277 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AVFoundation
+import WebKit
+import MobileCoreServices
+import YubiKit
+
+public class PlaylistMimeTypeDetector {
+    private(set) var mimeType: String = ""
+    private(set) var fileExtension: String = ""
+    
+    init(mimeType: String) {
+        self.mimeType = mimeType
+        
+        let mimeTypeMap = [
+            "video/webm": "webm",
```

Comment:
I thought webm is not going to be supported.

Reply:
It is not supported, but the mime-detector still detects if the video type is WebM. So we still need to detect if it is WebM in order to not play it.